samples/apps/copilot-chat-app/webapi/DeploymentTemplates/DeploySK-Existing-AzureOpenAI.ps1

```diff
@@ -51,12 +51,42 @@ param(
     [string]
     # SKU for the Azure App Service plan
     $AppServiceSku = "B1",
-    
+
+    [switch]
+    # Don't deploy Qdrant for memory storage - Use volatile memory instead
+    $NoQdrant,
+
+    [switch]
+    # Don't deploy Cosmos DB for chat storage - Use volatile memory instead
+    $NoCosmosDb,
+
+    [switch]
+    # Don't deploy Speech Services to enable speech as chat input
+    $NoSpeechServices,
+
     [switch]
     # Switches on verbose template deployment output
     $DebugDeployment
 )
 
+$jsonConfig = "
+{
+    `\`"name`\`": { `\`"value`\`": `\`"$DeploymentName`\`" },
+    `\`"endpoint`\`": { `\`"value`\`": `\`"$Endpoint`\`" },
+    `\`"apiKey`\`": { `\`"value`\`": `\`"$ApiKey`\`" },
+    `\`"completionModel`\`": { `\`"value`\`": `\`"$CompletionModel`\`" },
+    `\`"embeddingModel`\`": { `\`"value`\`": `\`"$EmbeddingModel`\`" },
+    `\`"plannerModel`\`": { `\`"value`\`": `\`"$PlannerModel`\`" },
+    `\`"packageUri`\`": { `\`"value`\`": `\`"$PackageUri`\`" },
+    `\`"appServiceSku`\`": { `\`"value`\`": `\`"$AppServiceSku`\`" },
+    `\`"deployQdrant`\`": { `\`"value`\`": $(If (!($NoQdrant)) {"true"} Else {"false"}) },
+    `\`"deployCosmosDB`\`": { `\`"value`\`": $(If (!($NoSpeechServices)) {"true"} Else {"false"}) },
+    `\`"deploySpeechServices`\`": { `\`"value`\`": $(If (!($NoSpeechServices)) {"true"} Else {"false"}) }
+}
+"
+
+$jsonConfig = $jsonConfig -replace '\s',''
```

Comment:
Although similar, none of the configs have the exact same fields.  Also, it is better NOT to have files for the configuration since they contain secrets (and a crashed script would leave them in plaintext on the disk).

Really, I moved the parameters to JSON strings so that I didn't have to change them at 3 places per script every time I needed to modify something. This JSON will never be instantiated. It will be passed to the az deployment commands as a string.  This is just more convenient than writing each of the parameters one by one for all the az deployment commands.

I think having JSON objects for this would actually make things more complicated.  It would for sure be a pain for me to deal with!